client/my-sites/sites/sites.jsx

```diff
@@ -27,6 +29,25 @@ export const Sites = React.createClass( {
 		path: React.PropTypes.string.isRequired
 	},
 
+	componentWillMount() {
+		// If we have a selected site already, and an un-registered
+		// path we want to handle, do a redirect rather than show
+		// the site site selector.
+		if ( this.props.selectedSite ) {
+			const siteFragment = getSiteFragment( page.current );
+			const path = this.props.path.toLowerCase().split( '?' )[ 0 ].split( '/' )[ 1 ];
```

Comment:
This logic already exists in `getHeaderText()`. What do you think about keeping this DRY with a reusable method?